### PR TITLE
implementing react-hook-form

### PR DESCRIPTION
**Pull request by filwas**

### CODE REVIEW

File: src/components/enumFaces/enumFaces.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  Dressing = "dressing",
+  Vegetables = "vegetables",
+  Egg = "egg",
+  Spreads = "spreads",
+  Serving = "serving",
+  Topping = "topping",
+}
+
+export interface PaniniFormData {
+  sandwichName: string[];
```
Comment (Sumick) [inline src/components/enumFaces/enumFaces.tsx:31]: **question:** why sandwich name is an array of strings?

File: src/components/screens/FormScreen.tsx [hunk 2]
```diff
@@ -3,35 +3,49 @@ import styles from "./FormScreen.module.css";
...
-  const [isBeingTurnedOff, setisBeingTurnedOff] = useState(false)
+  const formMethods = useForm<PaniniFormData>();
+  const { handleSubmit } = formMethods;
 
-  const handleOrder = () => {
-    setisBeingTurnedOff(true)
-  }
+  const onSubmit: SubmitHandler<PaniniFormData> = (data) => {
+    setisBeingTurnedOff(true);
+    console.log(data);
```
Comment (Sumick) [inline src/components/screens/FormScreen.tsx:19]: **nitpick:** Log to remove

File: src/components/screens/FormScreen.tsx [hunk 3]
```diff
@@ -3,35 +3,49 @@ import styles from "./FormScreen.module.css";
...
 
 function FormScreen() {
-  const [isBeingTurnedOff, setisBeingTurnedOff] = useState(false)
+  const formMethods = useForm<PaniniFormData>();
+  const { handleSubmit } = formMethods;
 
-  const handleOrder = () => {
-    setisBeingTurnedOff(true)
-  }
+  const onSubmit: SubmitHandler<PaniniFormData> = (data) => {
```
Comment (Sumick) [inline src/components/screens/FormScreen.tsx:17]: **nitpick:** data is not typed

File: src/components/selectorComponents/Checkbox.tsx [hunk 4]
```diff
@@ -3,38 +3,52 @@ import styles from "./Checkbox.module.css";
...
+
+  const toggleItemState = (index: number) => {
+    setItemStates((prevStates) => {
+      const newStates = [...prevStates];
+      newStates[index] = !prevStates[index];
+      return newStates;
+    });
+  };
+
+  const returnStates = itemStates.map((value, index) => value == true ? props.checkboxOptions[index] : "").filter(item => item != "");
```
Comment (Sumick) [inline src/components/selectorComponents/Checkbox.tsx:22]: **nitpick:** Not meaningful naming, it's not clear what returnStates are

**Review by Sumick — APPROVED** [review]
Looks very nice! Great job 💪 Im leaving few non blocking comments

File: src/components/enumFaces/enumFaces.tsx (hunk 1 above)
Comment (filwas) [reply src/components/enumFaces/enumFaces.tsx:31]: No real reason to be honest. Previously PaniniFormData was already formatted just like the payload, which was causing problems. I got frustrated so I made the object flat and changed all types to string arrays. I could change sandwichName to a single string, but it just looks consistent. ;)

File: src/components/screens/FormScreen.tsx (hunk 3 above)
Comment (filwas) [reply src/components/screens/FormScreen.tsx:17]: I believe the type for this data was created in the next PR

**Pull request merged**